commcare_connect/templates/opportunity/user_visit_details.html

```diff
@@ -0,0 +1,317 @@
+          <a :href="slides[modalCurrentIndex].src"
+             download="downloaded-image.jpg"
+             aria-label="Download Image">
+            <i class="fa-light fa-download text-brand-deep-purple"></i>
+          </a>
+        </button>
+      </div>
+
+      <!-- Thumbnails -->
+      <div class="relative w-full flex items-center justify-center gap-4 mt-4">
+        <button class="absolute -left-1 z-10 px-4.5 py-2 bg-white border-2 rounded-3xl border-brand-border-light"
+                @click="prevModal()"
+                :disabled="modalCurrentIndex === 0"
+                :class="{'opacity-50': modalCurrentIndex === 0}">
+          <i class="fa-light fa-arrow-left"></i>
+        </button>
+
+        <div class="flex justify-center gap-2 px-16 overflow-hidden">
+          <template x-for="(slide, idx) in modalThumbnails" :key="idx">
+            <div class="relative w-16 h-16 flex-shrink-0 cursor-pointer" @click="setModalImage(slides.indexOf(slide))">
+              <img
+                :src="slide.src"
+                :alt="slide.alt"
+                class="object-cover w-full h-full rounded-lg"
+              />
+              <div
+                x-show="slides.indexOf(slide) === modalCurrentIndex"
+                class="absolute inset-0 bg-black/40 rounded-lg pointer-events-none transition-all duration-300">
+              </div>
+            </div>
+          </template>
+        </div>
+
+
+        <button class="absolute -right-1 z-10 px-4.5 py-2 bg-white border-2 rounded-3xl border-brand-border-light"
+                @click="nextModal()"
+                :disabled="modalCurrentIndex >= slides.length - 1"
+                :class="{'opacity-50': modalCurrentIndex >= slides.length - 1}">
+          <i class="fa-light fa-arrow-right"></i>
+        </button>
+      </div>
+    </div>
+  </div>
+</div>
+
+
+<div class="flex justify-end gap-4 items-center h-14 bg-white rounded-lg shadow-sm"
+  x-data="{showRejectModal: false, showApproveModal: false}"
+  id="visit-actions"
+  hx-swap-oob="true">
+  <div class="flex gap-4">
+    <button class="button button-md outline-style" @click="showRejectModal = true; showApproveModal = false;">Reject</button>
+    {% if user_visit.opportunity.managed %}
+    <button class="button button-md primary-dark" @click="showRejectModal = false; showApproveModal = true;">Approve</button>
+    {% else %}
+    <button hx-post="{% url "opportunity:approve_visit" org_slug=request.org.slug pk=user_visit.id %}"
+      hx-swap="none" type="submit" class="button button-md positive-dark">Approve</button>
+    {% endif %}
+  </div>
+
+  <div x-cloak x-show="showRejectModal" x-transition.opacity class="modal-backdrop">
+    <div @click.away="showRejectModal = false" x-transition class="modal">
+      <div class="header flex justify-between items-center">
+        <h2 class="title">Reason for Rejection</h2>
+        <button @click="showRejectModal = false" class="button-icon" aria-label="Close">
```

Comment:
this button should be red, and the aria label should say "Reject"

Reply:
This button is for the `x` icon that closes the modal.